Source/WebCore/html/parser/AtomHTMLToken.h

```diff
@@ -67,39 +65,33 @@ class AtomHTMLToken {
 
     // Characters
 
-    const UChar* characters() const;
-    unsigned charactersLength() const;
+    Span<const UChar> characters() const;
     bool charactersIsAll8BitData() const;
 
     // Comment
 
     const String& comment() const;
     String& comment();
 
-    HasDuplicateAttribute hasDuplicateAttribute() const { return m_hasDuplicateAttribute; };
+    bool hasDuplicateAttribute() const { return m_hasDuplicateAttribute; }
 
 private:
-    HTMLToken::Type m_type;
-
     void initializeAttributes(const HTMLToken::AttributeList& attributes);
 
     AtomString m_name; // StartTag, EndTag, DOCTYPE.
-
     String m_data; // Comment
+    std::unique_ptr<DoctypeData> m_doctypeData; // DOCTYPE.
+    Vector<Attribute> m_attributes; // StartTag, EndTag.
 
     // We don't want to copy the characters out of the HTMLToken, so we keep a pointer to its buffer instead.
     // This buffer is owned by the HTMLToken and causes a lifetime dependence between these objects.
     // FIXME: Add a mechanism for "internalizing" the characters when the HTMLToken is destroyed.
-    const UChar* m_externalCharacters; // Character
-    unsigned m_externalCharactersLength; // Character
-    bool m_externalCharactersIsAll8BitData; // Character
-
-    std::unique_ptr<DoctypeData> m_doctypeData; // DOCTYPE.
-
-    bool m_selfClosing; // StartTag, EndTag.
-    Vector<Attribute> m_attributes; // StartTag, EndTag.
+    Span<const UChar> m_externalCharacters; // Character
 
-    HasDuplicateAttribute m_hasDuplicateAttribute { HasDuplicateAttribute::No };
+    HTMLToken::Type m_type;
```

Comment:
Can we make code more concise writing something like `using Type = HTMLToken::Type` in `AtomHTMLToken`?

Reply:
Done.